### DISCUSSION TITLE
For loop or While loop

### DISCUSSION THREAD

**Discussion by dpkhatua** [epochpic/epoch#489] [General]
Hii everyone,
Can anybody please suggest to me how to use a for loop or while loop in the input.deck. I want to work with a periodic structure and it is very painful to write the density manually.

Thank you.

**Comment by Status-Mirror** [comment]
The input deck doesn't support loops. The [maths parser](https://epochpic.github.io/documentation/code_details/maths_parser.html) can only read a few periodic functions (like `sin`).

You could write some code to automatically write your input deck, or you could write the density in another script and read the density in as a [binary file](https://epochpic.github.io/documentation/input_deck/binary_files.html).

Hope this helps,
Stuart

**Reply by dpkhatua** [reply]
Thank a lot!

DP
